### PR TITLE
Extended docs

### PR DESCRIPTION
**Pull request by pbbwfc**
* Created Documentation section with subsections covering different
aspects
* Simplified index page and referenced new sections
* Needed to update libraries as bug in Fake prevented correct build - see [fix](https://github.com/fsprojects/ProjectScaffold/issues/231)
* Changed generate.fsx to allow embedding output values (included fsi Evaluator  and added to calls to FSharp.Formatting)

### CODE REVIEW

**Comment by pbbwfc** [comment]
I hope this is OK and clear. Please just let me know if you want me to make any changes or if I have made any mistakes.
Thanks
Phil Brooks

**Comment by pbbwfc** [comment]
I just wanted to add that you can see the effect of the changes in [http://pbbwfc.github.io/ExcelProvider/](http://pbbwfc.github.io/ExcelProvider/)

File: docs/tools/generate.fsx.bak [hunk 1]
```diff
@@ -0,0 +1,145 @@
+// --------------------------------------------------------------------------------------
```
Comment (forki) [inline docs/tools/generate.fsx.bak]: please remove the bak files

**Comment by forki** [comment]
this is awesome!

**Comment by forki** [comment]
argh. 

somehow this breaks the build on my machine again.

image

@pbbwfc to get it out asap. can you please send a PR with the generated docs to the gh-pages branch? (just as temp solution)

@matthid any ideas why 600236a7fa64fd9b76405bd8c2cb606ef5064784 works for me and 349249a172eb6af1ca8de95fde354457a66dcb86 doesn't?

**Comment by matthid** [comment]
I think the reason might be that the ExcelProvider code is not compatible with a loaded Assembly that supports localization via resource assemblies.

In the context of `FSharp.Formatting` such a assembly is loaded (System.Web.Razor)

The problem is that ExcelProvider adds an incomatible global assembly handler here: https://github.com/fsprojects/ExcelProvider/blob/master/src/ExcelProvider/ExcelProvider.fs#L116
The handler simply tries to load the assembly which it cannot (it really should ignore calls it can't handle by returning null). 

It is not visible on travis because the runtime will not try to find a localized version as the assembly itself already has the correct localization so the runtime will not try to load additional assemblies there [1].

As a side-note: I think ExcelProvider should remove the event handler when it is no longer required or register a single global handler.

[1] https://msdn.microsoft.com/en-us/library/system.appdomain.assemblyresolve%28v=vs.110%29.aspx

**Comment by forki** [comment]
Can you please help me out with a pull request? I'd love to get these docs
out on the Web. Thanks

**Comment by pbbwfc** [comment]
Steffen,

 

Sorry I have not been online for a few hours.

 

Looks like things have moved on a bit – not sure what you want me to do. It looks like you have taken in the changes to the master branch but the docs on http://fsprojects.github.io/ExcelProvider/ have not been updated (and gh-pages branch hasn’t changed).

 

Have you tried running “build ReleaseDocs” to update the gh-pages branch or is that not working?

 

Regards

Phil Brooks

**Comment by forki** [comment]
There is a new pr which tries to fix the docs build. Will try that

**Comment by forki** [comment]
new docs are online. thx everybody

**Pull request merged**